portals/admin/src/main/webapp/source/src/app/components/AiVendors/AddEditAiVendor.jsx

```diff
@@ -0,0 +1,789 @@
+import {
+    MenuItem,
+    Typography,
+} from '@mui/material';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import FormControl from '@mui/material/FormControl';
+import API from 'AppData/api';
+import Alert from 'AppComponents/Shared/Alert';
+import Grid from '@mui/material/Grid';
+import Select from '@mui/material/Select';
+import { styled } from '@mui/material/styles';
+import TextField from '@mui/material/TextField';
+import ContentBase from 'AppComponents/AdminPages/Addons/ContentBase';
+import AIAPIDefinition from './AIAPIDefinition';
+
+const StyledSpan = styled('span')(({ theme }) => ({ color: theme.palette.error.dark }));
+
+const StyledHr = styled('hr')({ border: 'solid 1px #efefef' });
+
+const StyledContentBase = styled(ContentBase)({
+    '@global': {
+        '.MuiFormControl-root': {
+            marginTop: '20px',
+        },
+        '.MuiFormControl-root:first-of-type': {
+            marginTop: '0',
+        },
+    },
+});
+
+/**
+ * Reducer
+ * @param {JSON} state The second number.
+ * @returns {Promise}
+ */
+function reducer(state, newValue) {
+    const { field, value } = newValue;
+    switch (field) {
+        case 'name':
+        case 'apiVersion':
+        case 'description':
+        case 'apiDefinition':
+            return { ...state, [field]: value };
+        case 'model':
+        case 'promptTokenCount':
+        case 'completionTokenCount':
+        case 'totalTokenCount':
+            return {
+                ...state,
+                configurations: {
+                    ...state.configurations,
+                    metadata: state.configurations.metadata.map((item) => (
+                        item.attributeName === field
+                            ? { ...item, ...value }
+                            : item
+                    )),
+                },
+            };
+        case 'authHeader':
+        case 'authQueryParameter':
+        case 'connectorType':
+            return {
+                ...state,
+                configurations: {
+                    ...state.configurations,
+                    [field]: value,
+                },
+            };
+        case 'all':
+            return value;
+        default:
+            return state;
+    }
+}
+
+/**
+ * Render a list
+ * @returns {JSX} Header AppBar components.
+ */
+export default function AddEditAiVendor(props) {
+    const intl = useIntl();
+    const [saving, setSaving] = useState(false);
+    const { match: { params: { id } }, history } = props;
+    const inputSources = ['payload', 'header', 'queryParams'];
+    const [authSource, setAuthSource] = useState('authHeader');
+    const authSources = ['authHeader', 'authQueryParameter'];
+    const [validating, setValidating] = useState(false);
+    const [file, setFile] = useState(null);
+    const [initialState] = useState({
```

Comment:
Is the setInitialState ommited here on purpose, can we use a const instead

Reply:
setInitialState is omitted because changes are managed through a reducer, and useState is used to ensure the UI re-renders when the state changes